modules/dnn/src/dnn.cpp

```diff
@@ -1120,6 +1126,15 @@ static Ptr<BackendWrapper> wrapMat(int backendId, int targetId, cv::Mat& m)
         return Ptr<BackendWrapper>(new NgraphBackendWrapper(targetId, m));
 #else
         CV_Error(Error::StsNotImplemented, "This OpenCV version is built without support of Inference Engine + nGraph");
+#endif
+    }
+    else if (backendId == DNN_BACKEND_WEBNN)
+    {
+        CV_Assert(haveWebNN());
+#ifdef HAVE_WEBNN
```

Comment:
should `CV_Assert(haveWebNN());` be guarded by `HAVE_WEBNN`?

Reply:
fixed.